runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -1031,73 +896,52 @@ static iree_status_t iree_hal_hip_device_queue_alloca(
-        allocation_size, &buffer));
-
-    iree_status_t status = iree_hal_deferred_work_queue_enqueue_alloc(
-        device->work_queue, wait_semaphore_list, signal_semaphore_list, buffer);
-    if (iree_status_is_ok(status)) {
-      status = iree_hal_deferred_work_queue_issue(device->work_queue);
-    }
-    if (iree_status_is_ok(status)) {
-      *out_buffer = buffer;
-    } else {
-      iree_hal_hip_buffer_set_allocation_empty(buffer);
-      iree_hal_resource_release(&buffer->resource);
-    }
-    return status;
-  } else if (!iree_all_bits_set(params.type,
-                                IREE_HAL_MEMORY_TYPE_HOST_VISIBLE) &&
-             iree_hal_hip_allocator_isa(
-                 iree_hal_device_allocator(base_device))) {
-    iree_hal_buffer_t* buffer = NULL;
-
-    IREE_RETURN_IF_ERROR(iree_hal_hip_device_pepare_async_alloc(
-        device, params, allocation_size, &buffer));
+  IREE_TRACE_ZONE_BEGIN(z0);
 
-    iree_status_t status = iree_hal_deferred_work_queue_enqueue_alloc(
-        device->work_queue, wait_semaphore_list, signal_semaphore_list, buffer);
-    if (iree_status_is_ok(status)) {
-      status = iree_hal_deferred_work_queue_issue(device->work_queue);
-    }
-    if (iree_status_is_ok(status)) {
-      *out_buffer = buffer;
-    } else {
-      iree_hal_hip_buffer_set_allocation_empty(buffer);
-      iree_hal_resource_release(&buffer->resource);
-    }
-    return status;
-  }
+  iree_hal_hip_device_t* device = iree_hal_hip_device_cast(base_device);
 
   // NOTE: block on the semaphores here; we could avoid this by properly
   // sequencing device work with semaphores. The HIP HAL is not currently
   // asynchronous.
-  IREE_RETURN_IF_ERROR(iree_hal_semaphore_list_wait(wait_semaphore_list,
-                                                    iree_infinite_timeout()));
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(
+      z0, iree_hal_semaphore_list_wait(wait_semaphore_list,
+                                       iree_infinite_timeout()));
 
   // Allocate from the pool; likely to fail in cases of virtual memory
   // exhaustion but the error may be deferred until a later synchronization.
   // If pools are not supported we allocate a buffer as normal from whatever
   // allocator is set on the device.
-  iree_status_t status =
-      iree_hal_allocator_allocate_buffer(iree_hal_device_allocator(base_device),
-                                         params, allocation_size, out_buffer);
+  iree_status_t status = iree_ok_status();
+  if (device->supports_memory_pools &&
+      !iree_all_bits_set(params.type, IREE_HAL_MEMORY_TYPE_HOST_VISIBLE)) {
+    uint64_t device_num = 0;
+    if (queue_affinity) {
+      device_num = iree_math_count_trailing_zeros_u64(queue_affinity);
+      if (device_num > device->topology.count) {
+        IREE_TRACE_ZONE_END(z0);
+        return iree_make_status(
+            IREE_STATUS_INVALID_ARGUMENT,
+            "device affinity out of range, maximum device is %d",
```

Comment:
```suggestion
            "device affinity out of range, maximum device is %" PRIhsz,
```
don't cast to avoid using the right type

for longs and sizes and such use the inttypes.h macros
for iree_host_size_t and iree_device_size_t we have PRIhsz and PRIdsz respectively, but all other types can use the standard C ones